src/core/iomgr/iomgr.h

```diff
@@ -37,11 +37,22 @@
 /* gRPC Callback definition */
 typedef void (*grpc_iomgr_cb_func)(void *arg, int success);
 
+typedef struct grpc_iomgr_cb {
```

Comment:
Let's rename this grpc_iomgr_closure, and remove that type from fd_posix.h.

They both represent a callback paired with a parameter.
This makes the notify_on case in fd_posix.c actually trivial, and removes a use of is_ext_managed.